backend/common/src/main/java/ai/verta/modeldb/common/futures/InternalFuture.java

```diff
@@ -275,7 +275,13 @@ public static <U> InternalFuture<U> retriableStage(
         .get()
         .whenComplete(
             (value, throwable) -> {
-              boolean retryCheckerFlag = retryChecker.apply(throwable);
+              boolean retryCheckerFlag = false;
+              try {
+                retryCheckerFlag = retryChecker.apply(throwable);
+              } catch (Throwable e) {
+                promise.completeExceptionally(
+                    new ExecutionException("retryChecker threw an exception. Not retrying.", e));
```

Comment:
Wouldn't this make the promise complete twice (another time on line 305)? Maybe we have to return here?

Reply:
It can't actually complete twice (once it's been completed, the other calls are no-ops), but it definitely would be clearer to return here. 👍

Reply:
fixed.